### PR TITLE
change binning aggregation from "mean" to "sum"

### PR DESCRIPTION
**Pull request by jcharkow**
### **User description**
binning aggregation by mean leads to strange trends when have annotations that the annotation peaks are much higher than everything else. Changing to sum leads to more realistic plot.

Binning by mean (old):
image

Binning by sum (new):
image

I also changed the peakmap but these are not tested.

___

### **PR Type**
enhancement

___

### **Description**
- Changed the binning aggregation method from "mean" to "sum" to provide more realistic plots, especially when annotation peaks are significantly higher than other data points.
- Improved code readability by updating method signatures and formatting.
- Adjusted the logic for binning and peak annotations to align with the new aggregation method.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>_core.py</strong><dd><code>Update binning aggregation from mean to sum in plots</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pyopenms_viz/_core.py

<li>Changed binning aggregation method from "mean" to "sum" for intensity <br>calculations.<br> <li> Updated method signatures and formatting for better readability.<br> <li> Adjusted logic to handle binning and peak annotations.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/pyopenms_viz/pull/23/files#diff-328b7528522cebed85fb4aaf56353a3160e15ff53820eca51611fe1a7e870f65">+37/-45</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by jcharkow** [comment]
Possibly should allow user to customize the aggregation method?

**Comment by singjc** [comment]
I did a bit more digging and made some comparisons. I added a tolerance binning method, to bin mz's based on a fixed tolerance. I found the sturges and freedman binning methods sometimes don't work as well for either sparse or really dense spectrums. I also added an aggregation param to allow the user to aggregate by 'sum', 'mean' or 'max`. Based on my comparisons using the mz tolerance bining method (with tol = 1) and using max as the aggregation method seems to return spectra that closely matches the orignal raw spectra and is a lot faster for plotting (5.007686 seconds for raw vs 0.614527 seconds for max mz tol=1 binning)

image

File: nbs/alphatims_tutorial.ipynb
Comment (timosachsenberg) [inline nbs/alphatims_tutorial.ipynb:1]: what is this nb about?

**Comment by timosachsenberg** [comment]
just for reference: I used max in the past for similar reasons.

File: nbs/alphatims_tutorial.ipynb
Comment (jcharkow) [reply nbs/alphatims_tutorial.ipynb:1]: Sorry this is a rough notebook added by mistake to this PR. Will remove

Comment (singjc) [reply nbs/alphatims_tutorial.ipynb:1]: It's not added by mistake. It's a tutorial notebook for showing an example of loading bruker tdf DIA and DDA data using alphatims and showcasing plotting with pyopenms_viz.

The notebook was updated and added to this PR to reflect the changes with the spectrum binning.

**Review by singjc — APPROVED** [review]

**Pull request merged**